freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/dns/DnsV1Endpoint.java

```diff
@@ -75,4 +77,34 @@ void deleteDnsZoneBySubnetId(@QueryParam("environment") @NotEmpty String environ
     @ApiOperation(value = DnsOperationDescriptions.DELETE_DNS_RECORD_BY_FQDN, produces = MediaType.APPLICATION_JSON, notes = FreeIpaNotes.FREEIPA_NOTES,
             nickname = "deleteDnsRecordByFqdnV1")
     void deleteDnsRecordsByFqdn(@QueryParam("environment") @NotEmpty String environmentCrn, @QueryParam("fqdn") @NotEmpty List<String> fqdns);
+
+    @POST
+    @Path("record/a")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = DnsOperationDescriptions.ADD_DNS_A_RECORD_BY_FQDN, produces = MediaType.APPLICATION_JSON, notes = FreeIpaNotes.FREEIPA_NOTES,
+            nickname = "addDnsARecordV1")
+    void addDnsARecord(@Valid AddDnsARecordRequest request);
+
+    @DELETE
+    @Path("record/a")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = DnsOperationDescriptions.DELETE_DNS_A_RECORD, produces = MediaType.APPLICATION_JSON, notes = FreeIpaNotes.FREEIPA_NOTES,
+            nickname = "deleteDnsARecordV1")
+    void deleteDnsARecord(@QueryParam("environment") @NotEmpty String environmentCrn, @QueryParam("dnsZone") @NotEmpty String dnsZone,
+            @QueryParam("record") @NotEmpty String record);
+
+    @POST
+    @Path("record/cname")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = DnsOperationDescriptions.ADD_DNS_A_RECORD_BY_FQDN, produces = MediaType.APPLICATION_JSON, notes = FreeIpaNotes.FREEIPA_NOTES,
+            nickname = "addDnsCnameRecordV1")
+    void addDnsCnameRecord(@Valid AddDnsCnameRecordRequest request);
+
+    @DELETE
+    @Path("record/cname")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = DnsOperationDescriptions.DELETE_DNS_CNAME_RECORD, produces = MediaType.APPLICATION_JSON, notes = FreeIpaNotes.FREEIPA_NOTES,
+            nickname = "deleteDnsCnameRecordV1")
+    void deleteDnsCnameRecord(@QueryParam("environment") @NotEmpty String environmentCrn, @QueryParam("dnsZone") @NotEmpty String dnsZone,
+            @QueryParam("record") @NotEmpty String record);
 }
```

Comment:
Should we also add an API to query the a/cname records?

Reply:
we might add later. it wasn't requested, so I won't implement it in this PR